### ISSUE TITLE
Feature request: option to mark the tree as read-only

### ISSUE THREAD

**Issue by EdwinFLopez** [patosai/tree-multiselect.js#3]
Hi. Thanks for this component. It works great and it's very usable.

I've come to an scenario that requires the entire tree to be frozen for updates (a tipical crud). So far, I've come to solve the requirement by enclosing the component within a disabled fieldset and then unbinding the events on the span, something like this:

HTML:
```html
<fieldset id="read-only-tree" disabled="disabled">
   <select id="tree" multiselect="multiselect">...</select>
</fieldset>
```

Javascript:
```javascript
$(document).ready(function () {
                var options = {
                    allowBatchSelection: false
                };
                $("#tree").treeMultiselect(options);
                $("#read-only-tree span.remove-selected").unbind();
});
```

However, I find this kind of hackish, so having this functionality would be awesome.

Thanks again.

**Comment by patosai** [comment]
So it'd kind of be like, display-only?

**Comment by EdwinFLopez** [comment]
Correct Patrick. Thanks a lot for answering this soon.

**Comment by patosai** [comment]
The option is called 'freeze' in v1.14.4.Try it out.

**Comment by EdwinFLopez** [comment]
It's just perfect. Thank you very much! So awesome!.

**Comment by patosai** [comment]
Awesome sauce.

**Comment by EdwinFLopez** [comment]
Hi Patrick.
There's an issue when more than one select exists in a page. The options are propagated to the others.

**Comment by patosai** [comment]
You'd need to do treeMultiselect() your frozen one alone.

**Comment by EdwinFLopez** [comment]
Hmm... I don't follow.
This is what I got:

```html
<div class="container">
            <div class="row">
                <div class="col-md-12">
                    <select id="companyList" multiple="multiple">
                        <option value="one" data-section="bank1" selected="selected" data-index="3">One</option>
                        <option value="two" data-section="bank1" selected="selected" data-index="1">Two</option>
                        <option value="three" data-section="bank2" selected="selected" data-index="2">Three</option>
                        <option value="four" data-section="bank2">Four</option>
                        <option value="wow" data-section="bank2">Wow</option>
                    </select>
                </div>
            </div>

            <div class="row">
                <div class="col-md-12">
                        <select id="companyListSingle">
                            <option value="one" data-section="bank1" selected="selected">One</option>
                            <option value="two" data-section="bank1">Two</option>
                            <option value="three" data-section="bank2">Three</option>
                            <option value="four" data-section="bank2">Four</option>
                            <option value="wow" data-section="bank2">Wow</option>
                        </select>
                </div>
            </div>
        </div>
```

```javascript
            $(document).ready(function () {
                var options = {
                    allowBatchSelection: false
                };
                $("#companyList").treeMultiselect(options);
                $("#companyListSingle").treeMultiselect({"freeze":true});
            });
```

And when I modify the companyList one, the selected items don't have the [x] span anymore. Is this intended?

**Comment by patosai** [comment]
Oh that's interesting, I'll need to look into this more.

**Comment by EdwinFLopez** [comment]
Thanks Patrick. Awesome work.

**Comment by patosai** [comment]
Alright, try it now, I believe I've fixed this.

**Comment by EdwinFLopez** [comment]
It works as expected now, thank you Patrick for looking into this.

There's a strange behavior I've just noticed, and I don't know if it is intended or if it's worth another issue:

When the select tag doesn't have the multiple attribute set, the first element is always selected, however if the multiple attribute is set, there is not selection by default. Let me show you:

```html
<select id="TreeNone"> <!-- no multiple -->
                        <option value="one" data-section="bank1">One</option>
                        <option value="two" data-section="bank1">Two</option>
                        <option value="three" data-section="bank2">Three</option>
                        <option value="four" data-section="bank2">Four</option>
                        <option value="wow" data-section="bank2">Wow</option>
                </select>
<script type="text/javascript">
            $(document).ready(function () {
                $("#TreeNone").treeMultiselect();
            });
        </script>
```

![image](https://cloud.githubusercontent.com/assets/10159067/9790424/9325db0a-579b-11e5-80cf-fd7366980bbd.png)

Different result when the multiple attr is set:

```html
                    <select id="TreeMulti" multiple="multiple">
                        <option value="one" data-section="bank1">One</option>
                        <option value="two" data-section="bank1">Two</option>
                        <option value="three" data-section="bank2">Three</option>
                        <option value="four" data-section="bank2">Four</option>
                        <option value="wow" data-section="bank2">Wow</option>
                    </select>
<script type="text/javascript">
            $(document).ready(function () {
                $("#TreeMulti").treeMultiselect();
            });
        </script>
```

![image](https://cloud.githubusercontent.com/assets/10159067/9790471/d34632de-579b-11e5-908a-0dddecd06512.png)

Anyways, thank you for bringing this component, it's great.

**Comment by patosai** [comment]
Hmm that's an issue with turning the select into a multiple. I just tried it on a lone select, and using jQuery to set the `multiple` attribute autoselected the first option.

**Comment by patosai** [comment]
Actually, having the single option select sets the first option as the value automatically.

**Comment by EdwinFLopez** [comment]
Ok, understood as intended. I believe this is important to mention in the documentation. 
Thank you very much Patrick for all your help, I'm closing this enhancement.

**Comment by lovegupta** [comment]
Thanks Patrick. Your component is working wonders for us.
I just came across an issue so thought to first clarify here before raising it as an issue-

When i use freeze: true option then component gets disabled which is great but it shows parents selected while they don't have any child as selected.
Below is the code-

`$(document).ready(function() {
			$("#<%= id_namespace + Constants.FIELD_OA_CONDITIONS %> ").treeMultiselect({ 
				searchable: true,
				startCollapsed: true,
				freeze: true,
				searchParams: ['text', 'description']
			});
		});`

And it renders the output as 
![1](https://user-images.githubusercontent.com/42310549/46441233-e12ecb80-c733-11e8-823f-1ba63d98cd5a.JPG)
If I expand the individual trees
![2](https://user-images.githubusercontent.com/42310549/46441238-e55ae900-c733-11e8-991f-91eab393f84f.JPG)

So if you notice then nothing in second tree is selected still I have Down Payment checkbox as selected. While I checked the rendered HTML, the checkbox is actually not checked 
![3](https://user-images.githubusercontent.com/42310549/46441351-3ff44500-c734-11e8-8c6e-084d3570961c.jpg)
But it appears on UI that all children of Down Payment parent are checked. Is it a bug or it is how it is meant to work. It is causing confusion for end user because parent is checked means all children are selected.
Same is behavior in all browsers (I checked IE, Firefox, Chrome)

Thanks !
